apstra/api_iba_widgets.go

```diff
@@ -44,271 +35,131 @@ type IbaWidgetData struct {
 	MaxItems           *int
 	CombineGraphs      string
 	VisibleColumns     []string
-	Type               enum.IbaWidgetType
 	UpdatedBy          string
 }
 
-type rawIbaWidget struct {
-	AggregationPeriod  *int     `json:"aggregation_period,omitempty"`
-	OrderBy            string   `json:"orderby,omitempty"`
-	StageName          string   `json:"stage_name,omitempty"`
-	ShowContext        bool     `json:"show_context,omitempty"`
-	Description        string   `json:"description,omitempty"`
-	AnomalousOnly      bool     `json:"anomalous_only,omitempty"`
-	CreatedAt          *string  `json:"created_at,omitempty"`
-	SpotlightMode      bool     `json:"spotlight_mode,omitempty"`
-	UpdatedAt          *string  `json:"updated_at,omitempty"`
-	ProbeId            string   `json:"probe_id"`
-	Label              string   `json:"label"`
-	Filter             string   `json:"filter,omitempty"`
-	TimeSeriesDuration *int     `json:"time_series_duration,omitempty"`
-	DataSource         string   `json:"data_source,omitempty"`
-	MaxItems           *int     `json:"max_items,omitempty"`
-	CombineGraphs      string   `json:"combine_graphs,omitempty"`
-	VisibleColumns     []string `json:"visible_columns,omitempty"`
-	Type               string   `json:"type"`
-	Id                 ObjectId `json:"id,omitempty"`
-	UpdatedBy          string   `json:"updated_by,omitempty"`
-}
-
-func (o *rawIbaWidget) polish() (*IbaWidget, error) {
-	var created, updated time.Time
-
-	if o.CreatedAt != nil {
-		t, err := time.Parse("2006-01-02T15:04:05.000000+0000", *o.CreatedAt)
-		if err != nil {
-			return nil, fmt.Errorf("failure parsing create time %s - %w", *o.CreatedAt, err)
-		}
-		created = t
-	}
+var (
+	_ json.Marshaler   = new(IbaWidgetData)
+	_ json.Unmarshaler = new(IbaWidgetData)
+)
 
-	if o.UpdatedAt != nil {
-		t, err := time.Parse("2006-01-02T15:04:05.000000+0000", *o.UpdatedAt)
-		if err != nil {
-			return nil, fmt.Errorf("failure parsing create time %s - %w", *o.CreatedAt, err)
-		}
-		created = t
+func (i *IbaWidgetData) UnmarshalJSON(bytes []byte) error {
+	// TODO implement me
+	var raw struct {
+		AggregationPeriod  *int     `json:"aggregation_period,omitempty"`
+		OrderBy            string   `json:"orderby,omitempty"`
+		StageName          string   `json:"stage_name,omitempty"`
+		ShowContext        bool     `json:"show_context,omitempty"`
+		Description        string   `json:"description,omitempty"`
+		AnomalousOnly      bool     `json:"anomalous_only,omitempty"`
+		SpotlightMode      bool     `json:"spotlight_mode,omitempty"`
+		ProbeId            string   `json:"probe_id"`
```

Comment:
This could be `ObjectId` to prevent the type conversion below.

Reply:
Done